### PR TITLE
Add C++20 mathematical constants

### PR DESCRIPTION
**Pull request by jhelgert**
C++20 brings a couple of mathematical constants in the form of variable templates, defined in the [`<numbers>`](https://en.cppreference.com/w/cpp/numeric/constants) header. This PR adds those constants. Most of them are already available as C macros in `math.h`, but those macros aren't part of the C standard. 

I wasn't really sure how to handle variable templates properly in Cython, so I declared all constants as `const double` for the time being.

### CODE REVIEW

File: tests/run/cpp_stl_numbers.pyx [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# mode: run
+# tag: cpp20
```
Comment (da-woods) [inline tests/run/cpp_stl_numbers.pyx:2]:
```suggestion
# tag: cpp20, cpp
```

File: Cython/Includes/libcpp/numbers.pxd [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    const double pi
+    const double inv_pi
+    const double inv_sqrtpi
+    const double ln2
+    const double ln10
+    const double sqrt2
+    const double sqrt3
+    const double inv_sqrt3
+    const double egamma
+    const double phi
```
Comment (scoder) [inline Cython/Includes/libcpp/numbers.pxd:15]:
```suggestion
    const double phi
```

File: tests/run/cpp_stl_numbers.pyx [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    """
+    return pi
+
+def test_e():
+    """
+    >>> import math
+    >>> test_e() == math.e
+    True
+    """
+    return e
```
Comment (scoder) [inline tests/run/cpp_stl_numbers.pyx:21]:
```suggestion
    return e
```

**Review by scoder — COMMENTED** [review]
Line endings are missing at the end of the files.

**Comment by scoder** [comment]
Triggering a fresh CI run.

**Comment by scoder** [comment]
I'm not sure if the tests are all that good because they compare floating point constants from different sources. CPython doesn't use the C++ headers but the libc/math.h headers for defining them. It's rather unlikely that there will be differences (π as a float64 value is probably not up for debate any more) but not entirely impossible. We'll see if we get any problems on other platforms.

**Comment by scoder** [comment]
Thanks

**Pull request merged**